airflow/serialization/serialized_objects.py

```diff
@@ -393,7 +392,7 @@ def _serialize(cls, var: Any) -> Any:  # Unfortunately there is no support for r
         elif isinstance(var, Param):
             return cls._encode(cls._serialize_param(var), type_=DAT.PARAM)
         elif isinstance(var, XComArg):
-            return cls._encode(cls._serialize_xcomarg(var), type_=DAT.XCOM_REF)
+            return cls._encode(serialize_xcom_arg(var), type_=DAT.XCOM_REF)
```

Comment:
Kind of intended, since I don’t want to pollute the serialisation logic with different XComArg types. Maybe using another key would improve the result? Say

```
{"_type": "xcom_ref", "_val": {"task_id": "...", "key": "..."}}

{
    "_type": "xcom_ref",
    "_val": {
         "kind": "map",
         "arg": {"task_id": "...", "key": "..."},
         "callables": [...]
    }
}
```